src/components/navigation.tsx

```diff
@@ -17,17 +17,17 @@ const Navigation = ({ sections, currentSlug }: Props) => {
     }
   };
   const className = isOpen ? 'side-nav side-nav--open' : 'side-nav';
-  let flatSections: NavigationSectionItem[] = [];
-  Object.keys(sections).forEach((sectionKey: string) => {
-    flatSections = [...flatSections, ...sections[sectionKey]];
-  });
-
-  const currentSlugIndex: number = flatSections.findIndex(
-    (flatSection: NavigationSectionItem) => flatSection.slug === currentSlug
-  );
 
-  flatSections.forEach((item: NavigationSectionItem, index: number) => {
-    item.isDone = index < currentSlugIndex;
+  const readStatus: Map<NavigationSectionItem['slug'], boolean> = new Map();
+  // Mark section items as read up to the one currently open
+  let isRead = true;
```

Comment:
Nit: could use boolean type for simple “isRead” local variable?

Reply:
Sure. Added.